### ISSUE TITLE
Generation of global/core-excised Chandra spectra

### ISSUE THREAD

**Issue by DavidT3** [DavidT3/XGA#1265]
At the end of this task we should have the ability to make Chandra spectra in global apertures (i.e. from radius of zero to some user-defined outer radius), and to make them in 'core-excised' apertures (from some inner radius to some outer radius, both user-defined).

We'll use specextract, which will be lovely because it'll basically do everything we need. 

Some general notes/thoughts:
* We have to remember that, even though we care the most about extended sources, XGA is a completely general use tool - as such, for things like ARF weighting during spectrum generation we have to check the type of source being passed in and change the settings accordingly (you can see we do that for XMM in the spectral generation stage).
* The RMF weighting option is pointless and extremely computationally expensive (thanks @astrophysics-megan) - we can also probably cut down on the energy resolution of the RMF for more speed.
* The input regions have to be in Chandra detector coordinates (easy to achieve, we did the same for XMM and eROSITA).
* You can look through the XMM spectrum generation functions for inspiration and things to look out for.

The first tangible step for this task would be to create a 'spec' section in generate.ciao, then create an internal function (so with a '_' at the beginning of the name) called _chandra_spec_cmds or something - that is the same setup we use for XMM, and if we write _chandra_spec_cmds properly, it will be useful both for global/core-excised spectra **and** (later on) annular spectra.

When you look at the XMM spectrum functions you can ignore all of the bits which are todo with generating spectra within 'region' (so with outer_radius = 'region') - that is a legacy option left over from an older version of XGA and will be removed.

**Comment by DavidT3** [comment]
XSPEC scripts run, but cannot currently be read in to the spectrum and sources because the spectrum file paths aren't in the format required, and this version of XGA's spec fitting needs to split that information up and assemble a key to find the right spectrum in the source object.

**Comment by rz-wang** [comment]
I temporarily removed the `stacked_spectra` parameter from the `source.Tx` function call in the LT pipeline, as `source.Tx` doesn't recognize this argument and would raise an unexpected argument error. Since the LT function is a general-purpose function, not specific to Chandra, I’ll add the parameter back once I’ve finished my testing.

**Comment by rz-wang** [comment]
Need to re-check the region file generation.

**Comment by DavidT3** [comment]
@rz-wang  Given all the work you've done on this, I think it's safe to say it's in review now?

**Comment by rz-wang** [comment]
Yes. I changed it back from in review before because I wanted to test something on it.
